### PR TITLE
Files creation

### PR DESCRIPTION
**Pull request by maasencioh**

### CODE REVIEW

File: app/Controllers/Http/FileController.ts [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+    try {
+      const params = await request.validate(CreateFileValidator);
+      const file = await this.moveFromMultipart(params.file, params.filename);
+      return response.ok(file);
+    } catch (error) {
+      return response.badRequest({ errors: [error] });
+    }
+  }
+
+  private async moveFromMultipart(
```
Comment (maasencioh) [inline app/Controllers/Http/FileController.ts]: where you thinking something like this? @targos https://github.com/zakodium/adonis-datadrive/issues/12

File: addons/tequila/routes.ts [hunk 2]
```diff
@@ -76,7 +77,8 @@ Route.get(
       session.put('mylims.auth.method', 'tequila');
 
       return response.redirect(`${Env.get('FRONTEND_URL')}/eln`);
-    } catch (err) {
+      // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    } catch (err: any) {
```
Comment (targos) [inline addons/tequila/routes.ts]: please revert these and disable the rule in the tsconfig instead. `"useUnknownInCatchVariables": false`

File: app/Controllers/Http/FileController.ts (hunk 1 above)
Comment (targos) [reply app/Controllers/Http/FileController.ts]: Yes, but I was thinking of putting this inside adonis-datadrive directly.

File: addons/tequila/routes.ts (hunk 2 above)
Comment (targos) [reply addons/tequila/routes.ts]: https://github.com/mylims/mylims/pull/153/commits/08803627a28fc14e247f0e793627ca89458ccf5a is wrong.
I was asking to revert the change and update the tsconfig.json file, not eslint.

File: package.json [hunk 3]
```diff
@@ -10,6 +10,7 @@
     "eslint-fix": "npm run eslint -- --fix",
     "prepare": "node scripts/prepare.mjs && npm run tailwind:build",
     "prettier": "prettier --check .",
+    "prettier-graphql": "prettier --write front/src/generated/graphql.tsx",
```
Comment (targos) [inline package.json]: codegen should already handle this: https://github.com/mylims/mylims/blob/a2b3b3a9b12e8361e5506347a93fe063752791a4/front/codegen.yml#L5

Comment (maasencioh) [reply package.json]: I'm always having an error there, is enforcing to use 
```
blabla & {
  tnatna
}
```
over 
```
blablabla &
  {taonstoa}
```

File: addons/tequila/routes.ts (hunk 2 above)
Comment (maasencioh) [reply addons/tequila/routes.ts]: sorry, should be fine now

File: addons/file-sync/Commands/Sync.ts [hunk 4]
```diff
@@ -108,7 +108,7 @@ export default class Sync extends BaseCommand {
 
     try {
       await sync.walk();
-    } catch (err) {
+    } catch (err: any) {
```
Comment (targos) [inline addons/file-sync/Commands/Sync.ts]: should still remove these `any`

**Comment by targos** [comment]
```
Error: app/Components/tailwind-ui/overlays/Modal.tsx(92,26): error TS2339: Property 'showModal' does not exist on type 'HTMLDialogElement'.
Error: app/Components/tailwind-ui/overlays/Modal.tsx(94,26): error TS2339: Property 'close' does not exist on type 'HTMLDialogElement'.
```

These should be fixed if you update the components.

File: package.json (hunk 3 above)
Comment (targos) [reply package.json]: How can I reproduce the error? If I generate the file with `npm run codegen-watch`, some comments are removed but the format stays the same.

**Review by targos — APPROVED** [review]

**Pull request merged**